frb_rust/src/rust2dart.rs

```diff
@@ -129,8 +135,8 @@ impl<T: IntoDart> StreamSink<T> {
 
     /// Add data to the stream. Returns false when data could not be sent,
     /// or the stream has been closed.
-    pub fn add(&self, value: T) -> bool {
-        self.rust2dart().success(value)
+    pub fn add<V: IntoIntoDart<D> + TypeEquals<Other = T>, D: IntoDart>(&self, value: V) -> bool {
```

Comment:
nit: just a naive idea - does this work?

```suggestion
    pub fn add<D: IntoDart>(&self, value: T) -> bool where T: IntoIntoDart<D> {
```

Reply:
Good idea. Yes that worked, then we can get rid of the TypeEquals dependency